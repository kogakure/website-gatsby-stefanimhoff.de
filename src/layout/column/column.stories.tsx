import React from 'react';
import styled from 'styled-components';

import { Main } from '../main';

import { Column } from '.';

const Box = styled.div`
  align-items: center;
  background: rgba(25, 25, 25, 0.3);
  display: flex;
  justify-content: center;
`;

const rowProps = {
  marginBottom: true,
};

export default {
  component: Column,
  title: 'Layout/Column',
};

export const Default = () => (
  <Main>
    <Column {...rowProps}>
      <Box>Default</Box>
    </Column>
  </Main>
);

export const Fullsize = () => (
  <Main>
    <Column size="fullsize" {...rowProps} debug>
      <Box>fullsize</Box>
    </Column>
  </Main>
);

export const Regular = () => (
  <Main>
    <Column size="regular" {...rowProps} debug>
      <Box>regular</Box>
    </Column>
  </Main>
);

export const Narrow = () => (
  <Main>
    <Column size="narrow" {...rowProps} debug>
      <Box>narrow</Box>
    </Column>
  </Main>
);
