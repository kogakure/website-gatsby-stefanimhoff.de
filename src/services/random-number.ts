export const getRandomNumber = (max: number) =>
  String(Math.floor(Math.random() * Math.floor(max)));
