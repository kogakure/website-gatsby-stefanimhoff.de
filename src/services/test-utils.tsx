import React from 'react';
import { render, RenderOptions, RenderResult } from '@testing-library/react';

import { EmojifyProvider, LocalizedDateProvider } from '../contexts';

const customRender = (
  node: React.ReactElement,
  options?: RenderOptions
): RenderResult => {
  const rendered = render(
    <EmojifyProvider>
      <LocalizedDateProvider>{node}</LocalizedDateProvider>
    </EmojifyProvider>,
    options
  );

  return {
    ...rendered,
    rerender: (ui: React.ReactElement, rerenderOptions?: RenderOptions) =>
      customRender(ui, { container: rendered.container, ...options }),
  };
};

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
