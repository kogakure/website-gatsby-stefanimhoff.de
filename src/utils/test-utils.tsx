import React from 'react';
import { render, RenderOptions, RenderResult } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';

import { EmojifyProvider, LocalizedDateProvider } from '../contexts';
import { ThemeType } from '../typings/Theme';
import { lightTheme } from '../themes';

interface CustomTheme {
  theme: ThemeType;
}

type RenderOptionsWithTheme = RenderOptions & CustomTheme;

const customRender = (
  node: React.ReactElement,
  options: RenderOptionsWithTheme = { theme: lightTheme }
): RenderResult => {
  const { theme } = options;
  const rendered = render(
    <ThemeProvider theme={theme}>
      <EmojifyProvider>
        <LocalizedDateProvider>{node}</LocalizedDateProvider>
      </EmojifyProvider>
    </ThemeProvider>,
    options
  );

  return {
    ...rendered,
    rerender: (
      ui: React.ReactElement,
      rerenderOptions?: RenderOptionsWithTheme
    ) => customRender(ui, { container: rendered.container, ...options }),
  };
};

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
