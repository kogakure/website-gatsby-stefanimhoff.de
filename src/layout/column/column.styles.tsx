import styled, { css } from 'styled-components';

import { mediaQuery } from '../tokens';

import { ColumnProps } from './column';

export const variants = {
  fullsize: css`
    grid-column: 1 / -1;
  `,
  narrow: css`
    grid-column: 2 / -2;

    @media (${mediaQuery.tablet}) {
      grid-column: 6 / -6;
    }

    @media (${mediaQuery.desktop}) {
      grid-column: 7 / -7;
    }
  `,
  regular: css`
    grid-column: 2 / -2;

    @media (${mediaQuery.tablet}) {
      grid-column: 3 / -3;
    }

    @media (${mediaQuery.desktop}) {
      grid-column: 6 / -6;
    }
  `,
};

const Column = styled.div<ColumnProps>`
  ${({ variant }) => variant && variants[variant]}
  width: 100%;
`;

export const Styled = {
  Column,
};
