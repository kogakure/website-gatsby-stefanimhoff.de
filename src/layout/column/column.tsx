import * as React from 'react';

import { Styled, variants } from './column.styles';

export type ColumnVariants = keyof typeof variants;
export type ColumnProps = {
  variant?: ColumnVariants;
};

export const Column: React.FC<ColumnProps> = ({
  variant = 'fullsize',
  children,
  ...props
}) => (
  <Styled.Column variant={variant} {...props}>
    {children}
  </Styled.Column>
);
