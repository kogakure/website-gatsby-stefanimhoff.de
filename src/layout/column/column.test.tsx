import React from 'react';
import { axe } from 'jest-axe';

import { render } from '../../services/test-utils';

import { Column } from './column';

describe('Column', () => {
  test('renders correctly', () => {
    const { container } = render(<Column />);

    expect(container.firstChild).toMatchSnapshot();
  });

  test('is accessible', async () => {
    const { container } = render(<Column />);
    const results = await axe(container);

    expect(results).toHaveNoViolations();
  });
});
