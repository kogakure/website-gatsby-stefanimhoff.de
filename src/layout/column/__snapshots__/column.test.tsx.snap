// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Column renders correctly 1`] = `
.c0 {
  grid-column: 1 / -1;
  width: 100%;
}

<div
  class="c0"
/>
`;
